src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/dto/DraftReferralTest.kt

```diff
@@ -0,0 +1,31 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.integration.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
+import java.util.Date
+import java.util.UUID
+
+class DraftReferralTest : IntegrationTestBase() {
+
+  @Test
+  fun `create draft referral dto from empty referral`() {
+    val referral = Referral()
+    var draftReferral = DraftReferral(referral)
+
+    assertThat(draftReferral.id).isNull()
+    assertThat(draftReferral.created).isNull()
```

Comment:
i'm glad you wrote this test because it does raise an interesting point about if this draft referral is valid. 

should we enforce that draft referrals always have an id and created date by throwing a runtime exception if you attempt to create one without? e.g.

```
data class DraftReferral(
  val id: String,
  val created: String,
) {
  constructor(referral: Referral) : this(
    referral.id!!.toString(),
    DateTimeFormatter.ISO_INSTANT.format(referral.created!!.toInstant()),
  )
}
```

Reply:
I think you're right.  The draft referral needs both.